### PR TITLE
Also match weakly validated ETags

### PR DESCRIPTION
**Pull request by cchangwen**
https://stackoverflow.com/questions/51973120/where-does-the-w-in-an-etag-appear-from

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag#

### CODE REVIEW

File: modules/httpcache/httpcache.go [hunk 1]
```diff
@@ -58,6 +58,7 @@ func HandleGenericETagCache(req *http.Request, w http.ResponseWriter, etag strin
 func checkIfNoneMatchIsValid(req *http.Request, etag string) bool {
 	ifNoneMatch := req.Header.Get("If-None-Match")
 	if len(ifNoneMatch) > 0 {
+		ifNoneMatch = strings.TrimPrefix(ifNoneMatch, "W/")
 		for _, item := range strings.Split(ifNoneMatch, ",") {
 			item = strings.TrimSpace(item)
```
Comment (KN4CK3R) [inline modules/httpcache/httpcache.go]: Each `item` could have the `W/` prefix.

**Review by KN4CK3R — CHANGES_REQUESTED** [review]

**Review by KN4CK3R — APPROVED** [review]

File: modules/httpcache/httpcache.go [hunk 2]
```diff
@@ -59,7 +59,7 @@ func checkIfNoneMatchIsValid(req *http.Request, etag string) bool {
 	ifNoneMatch := req.Header.Get("If-None-Match")
 	if len(ifNoneMatch) > 0 {
 		for _, item := range strings.Split(ifNoneMatch, ",") {
-			item = strings.TrimSpace(item)
+			item = strings.TrimPrefix(strings.TrimSpace(item), "W/")
```
Comment (delvh) [inline modules/httpcache/httpcache.go]:
```suggestion
			item = strings.TrimPrefix(strings.TrimSpace(item), "W/") // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag#directives
```

**Review by delvh — APPROVED** [review]

**Comment by delvh** [comment]
Is that an enhancement or a bugfix?

**Comment by KN4CK3R** [comment]
I would say enhancement.

**Review by silverwind — APPROVED** [review]

**Pull request merged**